core/src/main/java/nu/yona/server/messaging/service/MessageService.java

```diff
@@ -285,16 +295,45 @@ private DtoManager getManager(Message messageEntity)
 	}
 
 	@Transactional
-	public void sendMessageAndFlushToDatabase(Message message, MessageDestinationDto destination)
+	public void sendMessageAndFlushToDatabase(Message message, UserAnonymizedDto toUser)
 	{
-		MessageDestination destinationEntity = getMessageDestination(destination.getId());
+		sendMessage(message, toUser);
+
+		MessageDestination destinationEntity = getMessageDestination(toUser.getAnonymousDestination().getId());
+		messageDestinationRepository.saveAndFlush(destinationEntity);
+	}
+
+	@Transactional
+	public void sendNamedMessageAndFlushToDatabase(Message message, User toUser)
+	{
+		sendNamedMessage(message, toUser);
+
+		MessageDestination destinationEntity = toUser.getNamedMessageDestination();
+		messageDestinationRepository.saveAndFlush(destinationEntity);
+	}
+
+	@Transactional
+	public void sendMessage(Message message, UserAnonymizedDto toUser)
+	{
+		MessageDestination destinationEntity = getMessageDestination(toUser.getAnonymousDestination().getId());
+
 		destinationEntity.send(message);
-		MessageDestination.getRepository().saveAndFlush(destinationEntity);
+
+		sendFirebaseNotification(message, toUser);
+	}
+
+	private void sendFirebaseNotification(Message message, UserAnonymizedDto toUser)
+	{
+		toUser.getDevicesAnonymized().stream().map(deviceAnonymized -> deviceAnonymized.getFirebaseInstanceId())
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://build.dev.yona.nu:9000/coding_rules#rule_key=squid%3AS1612)